hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldOperations.java

```diff
@@ -869,251 +869,251 @@ public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGeneric
-                writer.writeArrayOfNullableFloats(fieldName, record.getArrayOfNullableFloats(fieldName));
+                writer.writeArrayOfNullableFloat32s(fieldName, record.getArrayOfNullableFloat32s(fieldName));
             }
 
             @Override
             public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
                 writeArrayJsonFormatted(stringBuilder, record.readAny(fieldName), StringBuilder::append);
             }
         };
-        ALL[FieldKind.NULLABLE_DOUBLE.getId()] = new FieldKindBasedOperations() {
+        ALL[FieldKind.NULLABLE_FLOAT64.getId()] = new FieldKindBasedOperations() {
             @Override
             public Double readObject(GenericRecord genericRecord, String fieldName) {
-                return genericRecord.getNullableDouble(fieldName);
+                return genericRecord.getNullableFloat64(fieldName);
             }
 
             @Override
             public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord genericRecord, String fieldName) {
-                writer.writeNullableDouble(fieldName, genericRecord.getNullableDouble(fieldName));
+                writer.writeNullableFloat64(fieldName, genericRecord.getNullableFloat64(fieldName));
             }
 
             @Override
             public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
                 stringBuilder.append(readObject(record, fieldName));
             }
         };
-        ALL[FieldKind.ARRAY_OF_NULLABLE_DOUBLES.getId()] = new FieldKindBasedOperations() {
+        ALL[FieldKind.ARRAY_OF_NULLABLE_FLOAT64S.getId()] = new FieldKindBasedOperations() {
             @Override
             public Object readObject(GenericRecord genericRecord, String fieldName) {
-                return genericRecord.getArrayOfNullableDoubles(fieldName);
+                return genericRecord.getArrayOfNullableFloat64s(fieldName);
             }
 
             @Override
             public Object readIndexed(InternalGenericRecord record, String fieldName, int index) {
-                return record.getNullableDoubleFromArray(fieldName, index);
+                return record.getNullableFloat64FromArray(fieldName, index);
             }
 
             @Override
             public int hashCode(GenericRecord record, String fieldName) {
-                return Arrays.hashCode(record.getArrayOfNullableDoubles(fieldName));
+                return Arrays.hashCode(record.getArrayOfNullableFloat64s(fieldName));
             }
 
             @Override
             public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord record, String fieldName) {
-                writer.writeArrayOfNullableDoubles(fieldName, record.getArrayOfNullableDoubles(fieldName));
+                writer.writeArrayOfNullableFloat64s(fieldName, record.getArrayOfNullableFloat64s(fieldName));
```

Comment:
I don't think we need the `s` at the end. I would prefer `writeArrayOfNullableFloat64`

Reply:
@ihsandemir updated the pr. Can you check again ?